### PR TITLE
[Release] Release of vizro 0.1.21

### PR DESCRIPTION
**Pull request by maxschulz-COL**
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Review by petar-qb — APPROVED** [review]

**Review by lingyielia — APPROVED** [review]

File: vizro-core/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,22 @@ See the fragment files in the [changelog.d directory](https://github.com/mckinse
...
+## Changed
+
+- Improve page loading time for `AgGrid`, `Table` and `Figure`. ([#644](https://github.com/mckinsey/vizro/pull/644))
+
+- Optimize vizro templates `vizro_dark` and `vizro_light` for charts. ([#645](https://github.com/mckinsey/vizro/pull/645))
+
+## Fixed
+
+- Fix py.cafe docs integration by pinning `ruff` dependency for mkdocs plugin. ([#640](https://github.com/mckinsey/vizro/pull/640))
+
```
Comment (antonymilne) [inline vizro-core/CHANGELOG.md:27]:
```suggestion
```

Comment (antonymilne) [reply vizro-core/CHANGELOG.md:27]: This is just a docs change, not a package change.

File: vizro-core/src/vizro/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 __all__ = ["Vizro"]
 
-__version__ = "0.1.21.dev0"
+__version__ = "0.1.21"
```
Comment (antonymilne) [inline vizro-core/src/vizro/__init__.py:16]: Don't we need to bump the version in our mkdocs.yml also now? We were planning to add a point to the release checklist to do that.

**Review by antonymilne — APPROVED** [review]
Approving subject to the comments!

**Pull request closed without merging**